src/internal/document_api/mod.rs

```diff
@@ -1469,11 +1598,53 @@ mod tests {
             aes_value,
             &doc_id,
             with_keys,
-        )?;
+        )?
+        .into_edoc(DocumentHeader::new(doc_id.clone(), seg_id));
```

Comment:
Isn't `new` going to clone the input again to construct the object?

Reply:
In Rust this is implementation specific. This "new" looks like:
```rust 
fn new(document_id: DocumentId, segment_id: usize) -> DocumentHeader {
```
This means that the new function takes ownership of both arguments so if you want to use `doc_id` again in this function we have to give `new` its own copy.